package/com.unity.formats.usd/Runtime/Scripts/Behaviors/UsdAsset.cs

```diff
@@ -787,6 +787,11 @@ public string usdFullPath
     /// be destroyed and imported as a result.
     /// </summary>
     private void ApplyVariantSelectionState(Scene scene, UsdVariantSet variants) {
+      var primSource = variants.GetComponent<UsdPrimSource>();
+      if (!primSource) {
+        Debug.LogError("Null UsdPrimSource while applying variant selection state.");
+        return;
+      }
       var selections = variants.GetVariantSelections();
       var path = variants.GetComponent<UsdPrimSource>().m_usdPrimPath;
```

Comment:
You already have `primSource` above, I don't think you need to re-do `variants.GetComponent<UsdPrimSource>()`

Reply:
Good catch - fixed